src/libaktualizr/storage/invstorage.h

```diff
@@ -131,13 +131,16 @@ class INvStorage {
 
   virtual void storeRoot(const std::string& data, Uptane::RepositoryType repo, Uptane::Version version) = 0;
   virtual bool loadRoot(std::string* data, Uptane::RepositoryType repo, Uptane::Version version) = 0;
-  bool loadLatestRoot(std::string* data, Uptane::RepositoryType repo) {
+  bool loadLatestRoot(std::string* data, const Uptane::RepositoryType repo) {
```

Comment:
Why? `const` when passing by value is not forbidden, and actually makes some sense, just looks a bit weird to me. I think if we decide it's a good style, it's better apply it consistently (like in the other methods of INvStorage).

Reply:
I'm not a fan of these to be honest, I think the readability cost out-weights the benefits.

Reply:
I prefer using `const` when possible to make the intention of immutability clear. The reason why this one stands out is that it is the only function definition in the header file. All of the other definitions in the .cc files have `const`, but clang-tidy complains about using `const` for declarations in header files because it has no effect. The only part where it matters is the definition.

Reply:
The problem is that const for pass-by-value is not a meaningful info for the caller, so it's not relevant in the prototype. It's only useful for the inside of the function, where it's equivalent to a const variable.

But it's bad style (imo) to modify a pass-by-value parameter anyway (in that case, define another intermediary variable), so this const doesn't bring much info.

And when looking at a prototype, the brain usually expects a reference or pointer after a const declaration. It breaks the conventional style of most of C and C++ libraries (including the standard ones).

Plus in this case, it breaks uniformity, so it pops out and slows down readability.

Reply:
This case is already an outlier since it's a complete function definition in the header file instead of just a virtual function declaration. However, I appear to be in the minority here, and my opinion is not strong, so I'll undo it. Should I also undo the other `const RepositoryType` instances in the .cc files?

Reply:
I undid all of the unnecessary `const` pass-by-value instances.